### PR TITLE
Barnhark/improving cubic flux

### PR DESCRIPTION
**Pull request by kbarnhart**
This changes the two cubic components to be more general and use Taylor series expansion of as many terms as a user wants.

### CODE REVIEW

File: landlab/components/depth_dependent_taylor_soil_creep/hillslope_depth_dependent_taylor_flux.py [hunk 1]
```diff
@@ -10,12 +10,26 @@
...
 
     """
-    This component implements a depth-dependent cubic diffusion rule,
+    This component implements a depth-dependent Taylor series diffusion rule,
     combining concepts of Ganti et al. (2012) and Johnstone and Hilley (2014).
 
+    Hillslope sediment flux uses a Taylor Series expansion of the Andrews-
+    Bucknam formulation of nonlinear hillslope flux derived following following
+    Ganti et al., 2012 with a depth dependent component inspired Johnstone and
+    Hilley (2014).. The flux is given as:
```
Comment (gregtucker) [inline landlab/components/depth_dependent_taylor_soil_creep/hillslope_depth_dependent_taylor_flux.py]: extra .

File: landlab/components/depth_dependent_taylor_soil_creep/hillslope_depth_dependent_taylor_flux.py [hunk 2]
```diff
@@ -79,70 +97,70 @@ class DepthDependentCubicDiffuser(Component):
...
         dtmax = courant_factor * dx * dx / Demax
-    
+
     Where the courant factor is a user defined scale (default is 0.2)
-    
-    The DepthDependentCubicDiffuser has a boolean flag that permits a user
+
+    The DepthDependentTaylorDiffuser has a boolean flag that permits a user
     to be warned if timesteps are too large for the slopes in the model grid
     (if_unstable = 'warn') and a boolean flag that turns on dynamic timesteppping
```
Comment (gregtucker) [inline landlab/components/depth_dependent_taylor_soil_creep/hillslope_depth_dependent_taylor_flux.py]: extra p

File: landlab/components/depth_dependent_taylor_soil_creep/hillslope_depth_dependent_taylor_flux.py [hunk 3]
```diff
@@ -79,70 +97,70 @@ class DepthDependentCubicDiffuser(Component):
...
     >>> mynodes = mg.nodes[2, :]
-    
+
     Lets try to move the soil with a large timestep. Without dynamic time
-    steps, this gives a warning that we've exceeded the dynamic timestep size 
+    steps, this gives a warning that we've exceeded the dynamic timestep size
     and should use a smaller timestep. We could either use the smaller timestep,
-    or specify that we want to use adynamic timestep. 
-    
+    or specify that we want to use adynamic timestep.
```
Comment (gregtucker) [inline landlab/components/depth_dependent_taylor_soil_creep/hillslope_depth_dependent_taylor_flux.py]: a dynamic

File: landlab/components/depth_dependent_taylor_soil_creep/hillslope_depth_dependent_taylor_flux.py [hunk 4]
```diff
@@ -79,70 +97,70 @@ class DepthDependentCubicDiffuser(Component):
...
+    exceeded AND you have not selected dynamic timestepping with
+    dynamic_dt=True. This may lead to infinite and/or nan values for slope,
+    elevation, and soil depth. Consider using a smaller time step or dynamic
+    timestepping. The Courant condition recommends a timestep of
     0.004 or smaller.
-    
-    Now, we'll re-build the grid and do the same exapmle with dynamic timesteps. 
-    
+
+    Now, we'll re-build the grid and do the same exapmle with dynamic timesteps.
```
Comment (gregtucker) [inline landlab/components/depth_dependent_taylor_soil_creep/hillslope_depth_dependent_taylor_flux.py]: example

File: landlab/components/depth_dependent_taylor_soil_creep/hillslope_depth_dependent_taylor_flux.py (hunk 3 above)
Comment (gregtucker) [inline landlab/components/depth_dependent_taylor_soil_creep/hillslope_depth_dependent_taylor_flux.py]: a dynamic

File: landlab/components/depth_dependent_taylor_soil_creep/hillslope_depth_dependent_taylor_flux.py (hunk 2 above)
Comment (gregtucker) [inline landlab/components/depth_dependent_taylor_soil_creep/hillslope_depth_dependent_taylor_flux.py]: extra p

File: landlab/components/depth_dependent_taylor_soil_creep/hillslope_depth_dependent_taylor_flux.py (hunk 1 above)
Comment (gregtucker) [inline landlab/components/depth_dependent_taylor_soil_creep/hillslope_depth_dependent_taylor_flux.py]: extra .

File: landlab/components/depth_dependent_taylor_soil_creep/hillslope_depth_dependent_taylor_flux.py [hunk 5]
```diff
@@ -317,34 +349,45 @@ def soilflux(self, dt, dynamic_dt=False, if_unstable='pass', courant_factor=0.2)
...
+
     def _update_flux_topography_soil_and_bedrock(self):
         """Calculate soil flux and update topography. """
         #Calculate flux
-        self.flux[:] = -((self.K*self.slope
-                       + (self.K/(self.slope_crit**2)) * np.power(self.slope, 3))
-                        * (1.0 - np.exp(-self.H_link
+
+        slope_term = 0.0
+        s_over_scrit = self.slope/self.slope_crit
```
Comment (gregtucker) [inline landlab/components/depth_dependent_taylor_soil_creep/hillslope_depth_dependent_taylor_flux.py]: PEP8 has space around operators

**Comment by kbarnhart** [comment]
@gregtucker thanks for comments will fix and merge (conditional on tests passing).

**Pull request merged**